src/es/ravenmanga/src/eu/kanade/tachiyomi/extension/es/ravenmanga/RavenManga.kt

```diff
@@ -0,0 +1,186 @@
+
+    override val lang = "es"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.client.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 2)
+        .build()
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
+        .add("Referer", baseUrl)
+
+    private val json: Json by injectLazy()
+
+    override fun popularMangaRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun popularMangaSelector(): String = "div#div-diario figure, div#div-semanal figure, div#div-mensual figure"
+
+    override fun popularMangaNextPageSelector(): String? = null
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val elements = document.select(popularMangaSelector())
+
+        val mangas = elements.map { element ->
+            SManga.create().apply {
+                thumbnail_url = element.selectFirst("img")!!.attr("abs:src")
+                title = element.selectFirst("figcaption")!!.text()
+                setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+            }
+        }.distinctBy { it.url }
+
+        return MangasPage(mangas, false)
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga = throw UnsupportedOperationException("Not used!")
+
+    override fun latestUpdatesRequest(page: Int): Request = GET(baseUrl, headers)
+
+    override fun latestUpdatesSelector(): String = "section.flex > div.grid > figure"
+
+    override fun latestUpdatesNextPageSelector(): String? = null
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        thumbnail_url = element.selectFirst("img")!!.attr("abs:src")
+        title = element.selectFirst("figcaption")!!.text()
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            if (query.length > 1) return GET("$baseUrl/comics#$query", headers)
+            throw IOException("La búsqueda debe tener al menos 2 caracteres")
+        }
+        return GET("$baseUrl/comics?page=$page", headers)
+    }
+
+    override fun searchMangaSelector(): String = "section.flex > div.grid > figure"
+
+    override fun searchMangaNextPageSelector(): String = "nav > ul.pagination > li > a[rel=next]"
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val query = response.request.url.fragment ?: return super.searchMangaParse(response)
+        val document = response.asJsoup()
+        val mangas = parseMangaList(document, query)
+        return MangasPage(mangas, false)
+    }
+
+    private fun parseMangaList(document: Document, query: String): List<SManga> {
+        val docString = document.toString()
+        val mangaListJson = JSON_PROYECT_LIST.find(docString)?.destructured?.toList()?.get(0).orEmpty()
```

Comment:
Just to keep the English pattern.

```suggestion
        val mangaListJson = JSON_PROJECT_LIST.find(docString)?.destructured?.toList()?.get(0).orEmpty()
```